### PR TITLE
A large `T_TYPE` -> `type_t` deprecation.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi!

Done in a bulk:

1. Renamed all `T_FOO` into `foo_t`.
2. No top-level template parameter starts with `T_` now.
3. Current builds find with `T_DEPRECATED_` disabled (external clients would fail though).

Remains:

1. Deprecate that `T_DEPRECATED_()`.
2. Unify `typedef`-s into `using`-s.

Thanks,
Dima

### CODE REVIEW

**Comment by dkorolev** [comment]
Also, also in "remains": `BRIEF_OF_T<>`, and perhaps other top-level "macros" we have left.

File: Storage/container/one_to_one.h [hunk 1]
```diff
@@ -38,19 +38,19 @@ namespace storage {
...
-  using T_FORWARD_MAP = ROW_MAP<T_ROW, const T*>;
-  using T_TRANSPOSED_MAP = COL_MAP<T_COL, const T*>;
-  using T_REST_BEHAVIOR = rest::behavior::Matrix;
+  using row_t = sfinae::ENTRY_ROW_TYPE<T>;
+  using col_t = sfinae::ENTRY_COL_TYPE<T>;
+  using key_t = std::pair<row_t, col_t>;
+  using elements_map_t = std::unordered_map<key_t, std::unique_ptr<T>, CurrentHashFunction<key_t>>;
+  using forward_map_t = ROW_MAP<row_t, const T*>;
+  using transposed_map_t = COL_MAP<col_t, const T*>;
+  using t_rest_behavior = rest::behavior::Matrix;
```
Comment (mzhurovich) [inline Storage/container/one_to_one.h]: `rest_behavior_t`?

File: Storage/container/sfinae.h [hunk 2]
```diff
@@ -53,17 +53,17 @@ struct KEY_ACCESSOR_IMPL {};
 
 template <typename ENTRY>
 struct KEY_ACCESSOR_IMPL<ENTRY, false> {
```
Comment (mzhurovich) [inline Storage/container/sfinae.h]: Hm. Why we decided to use this style for struct names here?

File: Storage/rest/sfinae.h [hunk 3]
```diff
@@ -39,7 +39,7 @@ constexpr bool Has_T_BRIEF(char) {
 }
 
 template <typename T>
-constexpr auto Has_T_BRIEF(int) -> decltype(sizeof(typename T::T_BRIEF), bool()) {
+constexpr auto Has_T_BRIEF(int) -> decltype(sizeof(typename T::brief_t), bool()) {
```
Comment (mzhurovich) [inline Storage/rest/sfinae.h]: I suppose this should also be renamed.

**Comment by mzhurovich** [comment]
@dkorolev Wow, this is big :)
Since it compiles and passes top-level test, I'm merging it to simplify further development.

**Comment by dkorolev** [comment]
@mzhurovich , thanks for the review & merge, https://github.com/C5T/Current/pull/461 takes care of your comments.

**Pull request merged**